packages/jest-runtime/src/index.ts

```diff
@@ -764,6 +772,19 @@ class Runtime {
     this._currentlyExecutingModulePath = lastExecutingModulePath;
   }
 
+  private createScriptFromCode(scriptSource: string, filename: string) {
+    try {
+      return new Script(this.wrapCodeInModuleWrapper(scriptSource), {
+        displayErrors: true,
+        filename: this._resolver.isCoreModule(filename)
+          ? `jest-nodejs-core-${filename}`
+          : filename,
+      });
+    } catch (e) {
+      throw handlePotentialSyntaxError(e);
```

Comment:
Do we really need to catch it here? It's already handled when generating `scriptSource`, isn't it?

Reply:
the syntax error is thrown when `new Script` evaluates the the code, not during transform

Reply:
well, depending on the error. In this case, the tests are for `import` which is valid syntax, so babel passes it though, but node doesn't support it. So we need to catch both

Reply:
Thanks, makes sense to me now